pkg/clioptions/clusterdiscovery/cluster.go

```diff
@@ -381,3 +389,25 @@ func LoadConfig(state *ClusterState) (*ClusterConfiguration, error) {
 
 	return config, nil
 }
+
+// getFeatureGatesAPIServerMetrics extracts enabled feature gates from the API server metrics endpoint.
+// It returns a list of feature gate names that are enabled.
+func getFeatureGatesAPIServerMetrics(ctx context.Context, configClient configclient.Interface) ([]string, error) {
+	rsp, err := configClient.ConfigV1().RESTClient().Get().AbsPath("/metrics").Do(ctx).Raw()
+	if err != nil {
+		return nil, err
+	}
+	featureGates := sets.NewString()
+	lines := strings.Split(string(rsp), "\n")
+	re := regexp.MustCompile(`kubernetes_feature_enabled\{name="([^"]+)".*\} 1`)
+	for _, line := range lines {
+		if strings.HasPrefix(line, "kubernetes_feature_enabled{") && strings.HasSuffix(line, "} 1") {
+			matches := re.FindStringSubmatch(line)
+			if len(matches) == 2 {
+				featureGates.Insert(matches[1])
+			}
+		}
+	}
+	logrus.Infof("Enabled feature gates from API server metrics: %v", featureGates.List())
```

Comment:
This is logged 292x in https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/29952/pull-ci-openshift-origin-main-e2e-aws-csi/1950879489646923776/artifacts/e2e-aws-csi/openshift-e2e-test/build-log.txt

Please reduce the log level.